File: euler/010/problem10.cs
using System;
using System.Collections.Generic;

public class PFact
{
    static public void Main ()
    {
        const long TOP_NUM = 2000000;
        long num = 2;
        List <long> primes = new List<long>();
        primes.Add(num);
        long sum = primes[0];
        NUM: while ( num < TOP_NUM ){
            num++;
            double numsqrt = Math.Sqrt(num);
            foreach ( long p in primes) {
                if( p > numsqrt){
                    break;
                }
                if(num % p == 0){
                    goto NUM;
                }
            }
            primes.Add(num);
            sum += num;
        }
        Console.WriteLine(sum);

    }
}

File: euler/010/problem10.pl
use constant TOP_NUM => 2_000_000;
my $num = 2;
my @primes = ($num);
my $sum = $primes[0];
NUM: while ($num < TOP_NUM ){
  #prime if it does not divide evenly into seen primes
  $num++;
  my $sqrt = sqrt($num);
  for my $p (@primes) {
      last if $p > $sqrt;
      next NUM if $num % $p == 0;
  }
  push @primes,$num;
  $sum += $num;
}

print $sum,"\n";
